metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/EntityConverterUtils.java

```diff
@@ -0,0 +1,474 @@
+   * @param wikidataMap The prefLabel in Wikidata organization
+   * @param addToAltLabelMap The values that could not be added to Zoho prefLabel and should be
+   *        added to Zoho altLabel
+   * @return the new Zoho prefLabel value that should
+   */
+  public void mergePrefLabel(Map<String, List<String>> zohoMap,
+      Map<String, List<String>> wikidataMap, Map<String, List<String>> addToAltLabelMap) {
+
+    // go through all keys of the Wikidata map and add not equal values to the Zoho map
+    for (Map.Entry<String, List<String>> entry : wikidataMap.entrySet()) {
+      String key = entry.getKey();
+      if (!zohoMap.containsKey(key)) {
+        // currently the prefLabel is list of strings, but the data in Zoho and wikidata has always
+        // only one value
+        zohoMap.put(key, new ArrayList<String>(entry.getValue()));
+      } else {
+        //do not duplicate existing pref labels
+        List<String> unknownPrefLabels = new ArrayList<>(entry.getValue());
+        unknownPrefLabels.removeAll(zohoMap.get(key));
+        //add values to alt labels map 
+        addMissingValuesToMap(key, entry.getValue(), addToAltLabelMap);
+      }
+    }
+  }
+
+  /**
+   * This method is adding to the language map the missing values from the provided list, but
+   * avoiding dupplicated entries.
+   * 
+   * @param key the language key used in the map
+   * @param values the data to be added to the map
+   * @param languageMap the language map to which the provided values will be added
+   */
+  private void addMissingValuesToMap(String key, List<String> values,
+      Map<String, List<String>> languageMap) {
+    if (!languageMap.containsKey(key)) {
+      // add all labels to map
+      languageMap.put(key, new ArrayList<String>(values));
+    } else {
+      // add values from the list to the existing values in the map
+      List<String> mergedList = mergeStringLists(languageMap.get(key), values);
+      languageMap.remove(key);
+      languageMap.put(key, mergedList);
+    }
+  }
+
+  /**
+   * This method extends base list by elements of add list if it is not a duplicate
+   * 
+   * @param values
+   * @param newValues
+   * @return the list containing the sum of the two provided lists
+   */
+  @SuppressWarnings("unchecked")
+  public List<String> mergeStringLists(List<String> values, List<String> newValues) {
+    if(newValues == null){
+      return null;//nothing to do
+    }
+    if(values == null){
+      //add all values from the newValues
+      return new ArrayList<String>(newValues);
+    }else{
+      return ListUtils.sum(values, newValues);
+    } 
+  }
+
+  /**
+   * This method merges the two string arrays, but without removing duplicates
+   * 
+   * @param base The base string array
+   * @param add The additional string array
+   * @return the resulting string array
+   */
+  public String[] mergeStringArrays(String[] base, String[] add) {
+    if(base == null && add == null)
+      return null;
+    
```

Comment:
that is taken in account in the following call to mergeStringLists. However the code is invoked only if the base is not null (only than we can find a Wikidata Organization)

Reply:
No it's not... if one of them is null you get nullpointer exceptions. The method Arrays.asList() is not null-safe. And in a library method you'd expect these checks so that other code can safely use it.